src/api/app/controllers/webui/session_controller.rb

```diff
@@ -1,47 +1,48 @@
 class Webui::SessionController < Webui::WebuiController
   before_action :kerberos_auth, only: [:new]
 
+  before_action :session_creator, only: [:create]
+  before_action :authenticate, only: [:create]
+  before_action :check_user_active, only: [:create]
+
   skip_before_action :check_anonymous, only: [:create]
 
   def new
     switch_to_webui2
   end
 
   def create
-    user = User.find_with_credentials(params[:username], params[:password])
-
-    unless user
-      RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui,failure=unauthenticated value=1')
-      redirect_to(new_session_path, error: 'Authentication failed')
-      return
-    end
-
-    unless user.is_active?
-      RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui,failure=disabled value=1')
-      redirect_to(root_path, error: 'Your account is disabled. Please contact the administrator for details.')
-      return
-    end
-
-    User.session = user
-    session[:login] = user.login
-    Rails.logger.debug "Authenticated as user '#{user.login}'"
-    RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui value=1')
-
+    User.session = @session_creator.user
+    session[:login] = @session_creator.user.login
```

Comment:
Hm, those two lines look like things that should happen in `SessionControllerService::SessionCreator` to me.

Reply:
The `User.session` for sure, however I think don't think it belongs there. The `session` is a global from the controller  and I'm not sure how would be a good way to manipulate it without being too hacky.